### ISSUE TITLE
THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=405 error=11 : invalid argument

### ISSUE THREAD

**Issue by cristinasegalin** [pytorch/pytorch#15797]
Hello, 

I have 4 RTX 2080, cuda 10, pytorch 1.0.0 and when running code I keep getting this error
THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=405 error=11 : invalid argument

**Comment by ngimel** [comment]
Please run your code with CUDA_LAUNCH_BLOCKING=1 environment variable to give the exact location of failure.

**Comment by cristinasegalin** [comment]
I actually just solved it compiling pytorch from source. Might be useful for someone else having the same problem

**Comment by hill2hill** [comment]
> I actually just solved it compiling pytorch from source. Might be useful for someone else having the same problem

Excuse me ,I just meet the same problem like you.So did you fix it by compiling pytorch follwing the tutorial in 'conda' style? Thank you so much.
I also wanna know the version of pytorch , cuda and cudnn which work well on your 2080ti.Because I use a 2080ti too.

**Comment by Dev2022** [comment]
Please let us know in detail...how you have solved. Thanks :)

**Comment by duygusar** [comment]
I got this error though it is really awkward because I do run a network with CUDA with no problems at all and using the same set up/ even using the same terminal I run another network and I get this error, obviously has to do with pytorch but why I am able to run a network with pytorch and CUDa perfectly, while I can't run the another one????

**Comment by RifleZhang** [comment]
I have the same problem. There are two threads talking about this, but didn't find answers.
https://discuss.pytorch.org/t/a-error-when-using-gpu/32761
https://discuss.pytorch.org/t/a-error-when-using-gpu/32761/20

**Comment by YiruS** [comment]
I have the same issue. RTX2080, for CUDA 10.0 and pytorch 1.0.0. Anyone has solved the same problem please provide the information of the solution. It would help a lot of people. Thanks!

**Comment by ezyang** [comment]
Hi @RifleZhang , someone suggested installing a specific wheel. Did that resolve your problem?

Also, all of you, please try compiling PyTorch master from source and seeing if that resolves the issue.

**Comment by YiruS** [comment]
> I have the same issue. RTX2080, for CUDA 10.0 and pytorch 1.0.0. Anyone has solved the same problem please provide the information of the solution. It would help a lot of people. Thanks!

Okay I have solve the problem. You cannot directly install pytorch, instead “pip3 install -U https://download.pytorch.org/whl/cu100/torch-1.0.0-cp36-cp36m-linux_x86_64.whl” work for me.

**Comment by RizhaoCai** [comment]
> > I have the same issue. RTX2080, for CUDA 10.0 and pytorch 1.0.0. Anyone has solved the same problem please provide the information of the solution. It would help a lot of people. Thanks!
> 
> Okay I have solve the problem. You cannot directly install pytorch, instead “pip3 install -U [https://download.pytorch.org/whl/cu100/torch-1.0.0-cp36-cp36m-linux_x86_64.whl”](https://download.pytorch.org/whl/cu100/torch-1.0.0-cp36-cp36m-linux_x86_64.whl%E2%80%9D) work for me.

Thank you very much. I follow your advice and I succeed. But there is a little difference. My device is NVIDIA Tesla T4. And I use python 3. So I change the command:
$ wget https://download.pytorch.org/whl/cu100/torch-1.0.0-cp35-cp35m-linux_x86_64.whl
$ python3.5 -m pip install torch-1.0.0-cp35-cp35m-linux_x86_64.whl

The previous pytorch was installed vi pip directly: $ pip3 install torch.
Your solution works for me. Thank you very much

**Comment by ShijianXu** [comment]
Hi, still have a problem.

I used the solution you provided and when run: `torch.version.cuda`, it will show `10.0.130`. However, if I continue to install torchvision via `pip3 install torchvision`, it will show this:

> Collecting torchvision
https://files.pythonhosted.org/packages/fb/01/03fd7e503c16b3dc262483e5555ad40974ab5da8b9879e164b56c1f4ef6f/torchvision-0.2.2.post3-py2.py3-none-any.whl
Collecting six (from torchvision)
https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting numpy (from torchvision)
https://files.pythonhosted.org/packages/c1/e2/4db8df8f6cddc98e7d7c537245ef2f4e41a1ed17bf0c3177ab3cc6beac7f/numpy-1.16.3-cp36-cp36m-manylinux1_x86_64.whl
Collecting pillow>=4.1.1 (from torchvision)
https://files.pythonhosted.org/packages/d2/c2/f84b1e57416755e967236468dcfb0fad7fd911f707185efc4ba8834a1a94/Pillow-6.0.0-cp36-cp36m-manylinux1_x86_64.whl
Collecting torch (from torchvision)
https://files.pythonhosted.org/packages/31/ca/dd2c64f8ab5e7985c4af6e62da933849293906edcdb70dac679c93477733/torch-1.0.1.post2-cp36-cp36m-manylinux1_x86_64.whl
Installing collected packages: six, numpy, pillow, torch, torchvision
Successfully installed numpy-1.16.3 pillow-6.0.0 six-1.12.0 torch-1.0.1.post2 torchvision-0.2.2.post3

Apparently, it install a `torch` again. And when I run `torch.version.cuda` again, it will return to `9.0.176`. As a result, the error `THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=405 error=11 : invalid argument` still exists. This is frustrated.

Do you know how to solve it?

**Comment by RizhaoCai** [comment]
> Hi, still have a problem.
> I used the solution you provided and when run: torch.version.cuda, it will show 10.0.130. However, if I continue to install torchvision via pip3 install torchvision, it will show this:
> 
> Collecting torchvision
> https://files.pythonhosted.org/packages/fb/01/03fd7e503c16b3dc262483e5555ad40974ab5da8b9879e164b56c1f4ef6f/torchvision-0.2.2.post3-py2.py3-none-any.whl
> Collecting six (from torchvision)
> https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
> Collecting numpy (from torchvision)
> https://files.pythonhosted.org/packages/c1/e2/4db8df8f6cddc98e7d7c537245ef2f4e41a1ed17bf0c3177ab3cc6beac7f/numpy-1.16.3-cp36-cp36m-manylinux1_x86_64.whl
> Collecting pillow>=4.1.1 (from torchvision)
> https://files.pythonhosted.org/packages/d2/c2/f84b1e57416755e967236468dcfb0fad7fd911f707185efc4ba8834a1a94/Pillow-6.0.0-cp36-cp36m-manylinux1_x86_64.whl
> Collecting torch (from torchvision)
> https://files.pythonhosted.org/packages/31/ca/dd2c64f8ab5e7985c4af6e62da933849293906edcdb70dac679c93477733/torch-1.0.1.post2-cp36-cp36m-manylinux1_x86_64.whl
> Installing collected packages: six, numpy, pillow, torch, torchvision
> Successfully installed numpy-1.16.3 pillow-6.0.0 six-1.12.0 torch-1.0.1.post2 torchvision-0.2.2.post3
> 
> Apparently, it install a torch again. And when I run torch.version.cuda again, it will return to 9.0.176. As a result, the error THCudaCheck FAIL file=/pytorch/aten/src/THC/THCGeneral.cpp line=405 error=11 : invalid argument still exists. This is frustrated.
> Do you know how to solve it?

Could you try to avoid using torchvision and keep torch.version.cuda->10.0.130. Then run the code to see if it is the problem brought by the torchvision. 


Or you can try to install torchvision from source (https://github.com/pytorch/vision) to keep torch from being reinstalled.

**Comment by JacobYuan7** [comment]
@cristinasegalin, what do you mean by 'compiling pytorch from source'?

**Comment by JacobYuan7** [comment]
Excuse me, I have a question. @cristinasegalin  what do you mean by 'compiling pytorch from source'?

**Comment by Yuriscu** [comment]
I met the same problem when my environment are pytorch0.4.1+cuda9.1 with RTX2080ti, and I solved the problem by applying the following environment: ptorch1.0+cuda10.0+RTX2080ti  .

**Comment by cobbwho** [comment]
hi, i have the same problem. I think the installed pytorch version does not match the actual cuda and cudnn versions. When I switched to the same cudnn version of pytorch, it's ok.

**Comment by zhouyuan888888** [comment]
I have successfully solved this problem by updating the cuda8.0 to cuda10.0.

**Comment by sharibox** [comment]
sudo rmmod nvidia_uvm

**Comment by MADHAVAN001** [comment]
I faced the same issue with PyTorch 1.0.1 + CUDA 10.1.

Reinstalling torch solves it for me.

**Comment by jbmaxwell** [comment]
Just rebooting solved it, in my case.

**Comment by wzdutd** [comment]
I solved this problem just following YiruS and RizhaoCai. 
RTX Titan + cuda 10.0!
Thank you!

**Comment by yuanzhedong** [comment]
I think it's due to cuda version and pytorch version mismatch, I got this error because I have cuda 10 and pytorch 1.5 but 1.5 only support 10.1 and 10.2 https://pytorch.org/get-started/previous-versions/

**Comment by sytelus** [comment]
FYI - in my case this error was occuring because of mismatch in number of classes in dataset and model logits. For example, if you try to use cifar10 model with cifar100 you will get this error.

**Comment by mianasbat** [comment]
I got this error when I was running a pytorch application in docker.     
I was using base image `nvidia/cuda:11.1.1-devel-ubuntu18.04` with `torch 1.4.0`
I changed the base image to `nvidia/cuda:10.2-devel-ubuntu18.04` with same torch version and the error disappeared.

**Comment by HuXia7157** [comment]
> I actually just solved it compiling pytorch from source. Might be useful for someone else having the same problem

How to compile pytorch from source?

**Comment by jichengyuan** [comment]
torch.backends.cudnn.benchmark = False works for me.
